quickwit/quickwit-indexing/src/actors/merge_executor.rs

```diff
@@ -367,6 +367,7 @@ impl MergeExecutor {
         let mut merged_index = Index::open(controlled_directory.clone())?;
         ctx.record_progress();
         merged_index.set_tokenizers(get_quickwit_tokenizer_manager().clone());
+        merged_index.set_fast_field_tokenizers(get_quickwit_fastfield_normalizer_manager().clone());
```

Comment:
The fast field tokenizers are useful only when indexing and merging right?

I saw two places where we don't set the fast field tokenizers:
- https://github.com/quickwit-oss/quickwit/blob/d7d4990133cbd4f0f2953d2b444f587abb8eb489/quickwit/quickwit-indexing/src/actors/merge_executor.rs#L515
- https://github.com/quickwit-oss/quickwit/blob/d7d4990133cbd4f0f2953d2b444f587abb8eb489/quickwit/quickwit-search/src/leaf.rs#L121

Reply:
Yes, but also I'm not sure. I'm not a fan of having two tokenizers managers, it makes things complicated and doesn't solve anything currently

Reply:
Let's set it both tokenizer manager in all sites without thinking about whether it is useful or not.

> I'm not a fan of having two tokenizers managers, it makes things complicated and doesn't solve anything currently

You are being unfair. You can say you would have preferred another solution, but you cannot say it is not solving a problem.

Let me repeat what I explained several times in PRs:
Before this addition we had one tokenizer manager that was targetting both indexing and fast fields. You introduced a tokenizer called `lowercase` for fast fields.

Now here is the problem I explained two or three times...
We had tokenizers called:
- `default`: splitting + lowercasing
- `lowercase`: not splitting + lowercasing
- `en_stem`: splitting + stemming.

This is quite confusing for a user isn't it? Normalizer and tokenizer are living in the same namespace with no hint in their names of who is doing what.

Another solution would have been to give more explicit names to tokenizers.
(note you did not do that)

I prefer to avoid any footgun and have two tokenizer managers.
You may argue that users may still populate the tokenizer manager correctly...
Or that the possibly colliding names of the tokenizer can be confusing...
But with the default settings of tantivy, it does remove a massive footgun.

Reply:
Sorry, I think I my statement wasn't clear, I was only talking about tantivy's two tokenizer API, not separating tokenizer and normalizers.

I agree having `normalizer` and `tokenizer` separated in quickwit is good and solves the problem of mixing them. The validation of them and mapping to the actual tokenizer name (via `get_name`) is done on quickwit side, so they are separated in the API and can't be mixed. In that context the tantivy's two tokenizer API doesn't solve anything or provide a functionality that is required. I see two tokenizer manager as a bigger footgun, than internally mapping a tokenizer to a normalizer or the other way around (this is not a quickwit user API anymore)